src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -0,0 +1,229 @@
+    Props(new WorkflowSubmissionActor(dataSource, methodRepoDAO, googleServicesDAO, executionServiceDAO, batchSize, credential, pollInterval))
+  }
+
+  sealed trait WorkflowSubmissionMessage
+  case object LookForWorkflows extends WorkflowSubmissionMessage
+  case class SubmitWorkflowBatch(workflowIds: Seq[Long]) extends WorkflowSubmissionMessage
+
+}
+
+class WorkflowSubmissionActor(val dataSource: SlickDataSource,
+                              val methodRepoDAO: MethodRepoDAO,
+                              val googleServicesDAO: GoogleServicesDAO,
+                              val executionServiceDAO: ExecutionServiceDAO,
+                              val batchSize: Int,
+                              val credential: Credential,
+                              val pollInterval: FiniteDuration) extends Actor with WorkflowSubmission with LazyLogging {
+
+  import context._
+
+  scheduleNextWorkflowQuery
+
+  override def receive = {
+    case LookForWorkflows =>
+      getUnlaunchedWorkflowBatch() pipeTo self
+    case SubmitWorkflowBatch(workflowIds) =>
+      submitWorkflowBatch(workflowIds) pipeTo self
+
+    case Status.Failure(t) =>
+      logger.error(t.getMessage)
+      scheduleNextWorkflowQuery
+  }
+
+  def scheduleNextWorkflowQuery: Cancellable = {
+    system.scheduler.scheduleOnce(pollInterval, self, LookForWorkflows)
+  }
+}
+
+trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths {
+  val dataSource: SlickDataSource
+  val methodRepoDAO: MethodRepoDAO
+  val googleServicesDAO: GoogleServicesDAO
+  val executionServiceDAO: ExecutionServiceDAO
+  val batchSize: Int
+  val credential: Credential
+  val pollInterval: FiniteDuration
+
+  import dataSource.dataAccess.driver.api._
+
+  //Get a blob of unlaunched workflows, flip their status, and queue them for submission.
+  def getUnlaunchedWorkflowBatch()(implicit executionContext: ExecutionContext): Future[WorkflowSubmissionMessage] = {
+    val unlaunchedWfOptF = dataSource.inTransaction { dataAccess =>
+      //grab a bunch of unsubmitted workflows
+      dataAccess.workflowQuery.findUnsubmittedWorkflows().take(batchSize).result map { wfRecs =>
+        if (wfRecs.nonEmpty) wfRecs
+        else {
+          //they should also all have the same submission ID
+          val wfsWithASingleSubmission = wfRecs.filter(_.submissionId == wfRecs.head.submissionId)
+          //instead: on update, if we don't get the right number back, roll back the txn
+          //TODO: biden has an incoming change here, with optimistic locking
+          dataAccess.workflowQuery.batchUpdateStatus(wfsWithASingleSubmission.map(_.id), WorkflowStatuses.Launching)
+          wfsWithASingleSubmission
+        }
+      }
+    }
+
+    //if we find any, next step is to submit them. otherwise, look again.
+    unlaunchedWfOptF.map {
+      case workflowRecs if workflowRecs.nonEmpty => SubmitWorkflowBatch(workflowRecs.map(_.id))
+      case _ => LookForWorkflows
```

Comment:
there should be a message to schedule next workflow (that uses the poll interval) and use it here. The way it stands, this will be a tight busy wait loop.